packages/uni_app/lib/controller/fetchers/library_occupation_fetcher.dart

```diff
@@ -1,26 +1,54 @@
-import 'package:uni/controller/fetchers/session_dependant_fetcher.dart';
-import 'package:uni/controller/networking/network_router.dart';
-import 'package:uni/controller/parsers/parser_library_occupation.dart';
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:http/http.dart';
 import 'package:uni/model/entities/library_occupation.dart';
-import 'package:uni/model/entities/session.dart';
 
 /// Fetch the library occupation from Google Sheets
-class LibraryOccupationFetcherSheets implements SessionDependantFetcher {
-  @override
-  List<String> getEndpoints(Session session) {
-    const baseUrl = 'https://docs.google.com/spreadsheets/d/';
-    const sheetId = '1gZRbEX4y8vNW7vrl15FCdAQ3pVNRJw_uRZtVL6ORP0g';
-    const url =
-        '$baseUrl$sheetId/gviz/tq?tqx=out:json&sheet=MANUAL&range=C2:E7&tq=SELECT+C,E';
-    return [url];
+class LibraryOccupationFetcher {
+  String baseUrl = 'https://webapi.affluences.com/api/fillRate?';
+
+  Map<String, int> floorMaxSeats = {
+    'BruV6IlujdwAe1': 72,
+    'cEhyzJZvC5nHSr': 114,
+    'iceVfgwZWaZRhV': 114,
+    '1yLPz9X0CNsg27': 114,
+    'keu1j5zERlQn90': 40,
+    'bY7K1v43HiAq55': 90,
+  };
+
+  Future<LibraryOccupation> getLibraryOccupation() async {
+    final libraryOccupation = LibraryOccupation(0, 0);
+
+    for (var i = 0; i < floorMaxSeats.entries.length; i++) {
+      final floor = floorMaxSeats.entries.elementAt(i);
+
+      final url = Uri.parse(baseUrl).replace(
+        queryParameters: {
+          'token': floor.key,
+        },
+      );
+
+      final response = await http.get(url);
```

Comment:
You could use `floorMaxSeats.forEach((key, value)...` here and cut two `ementAt`

Reply:
it needs to be `await Future.all(floorMaxSeats.map((key, value)...` for the method to work properly, but yes, I agree!